core/src/main/scala/kafka/server/AdminManager.scala

```diff
@@ -700,4 +702,176 @@ class AdminManager(val config: KafkaConfig,
     val readOnly = !DynamicBrokerConfig.AllDynamicConfigs.contains(name)
     new DescribeConfigsResponse.ConfigEntry(name, valueAsString, source, isSensitive, readOnly, synonyms.asJava)
   }
+
+  private def entityToSanitizedUserClientId(entity: QuotaEntity): (Option[String], Option[String]) = {
+    if (entity.entries.isEmpty)
+      throw new InvalidRequestException("Invalid empty quota entity")
+
+    var user: Option[String] = None
+    var clientId: Option[String] = None
+    entity.entries().asScala.foreach { case (entityType, entityName) =>
+      val sanitizedEntityName = Some(Sanitizer.sanitize(entityName))
+      entityType match {
+        case QuotaEntity.USER => user = sanitizedEntityName
+        case QuotaEntity.CLIENT_ID => clientId = sanitizedEntityName
+        case _ => throw new InvalidRequestException(s"Unhandled quota entity type: ${entityType}")
+      }
+      if (entityName == "")
```

Comment:
I'd really rather not see the double equals used with strings.  I know Scala has special semantics for this, but it always looks wrong to people familiar with Java.  Maybe test `isEmpty` instead?

Reply:
Done.